### ISSUE TITLE
De-registering Tentacle does not remove server from config file

### ISSUE THREAD

**Issue by matt-richardson** [OctopusDeploy/Issues#3246]
Tentacle version 3.11.4+Branch.master.Sha.7c4e96df2e9b16d55e4abe4190ad18b1f454630f

Appears related to #873. Raised as a discussion point to start with.

After running the following series of commands, the Tentacle is not accessible:

```
# install
C:\Windows\system32\msiexec.exe  /i C:\Octopus\Installer\Tentacle.msi /quiet /l*v c:\temp\tests\Integration\Scenarios\01_Install\logs\Tentacle.Installer.log
# setup instance as listening
C:\Program Files\Octopus Deploy\Tentacle\Tentacle.exe create-instance --instance Tentacle --config C:\Octopus\Tentacle\Tentacle.config --console
C:\Program Files\Octopus Deploy\Tentacle\Tentacle.exe configure --instance Tentacle --home C:\Octopus\Tentacle --console
C:\Program Files\Octopus Deploy\Tentacle\Tentacle.exe configure --instance Tentacle --app C:\Applications --console
C:\Program Files\Octopus Deploy\Tentacle\Tentacle.exe new-certificate --instance Tentacle --console
C:\Program Files\Octopus Deploy\Tentacle\Tentacle.exe service --install --stop --instance Tentacle--console
C:\Windows\system32\sc.exe config "OctopusDeploy Tentacle" start= disabled
C:\Program Files\Octopus Deploy\Tentacle\Tentacle.exe configure --instance Tentacle --port 10933 --console
C:\Program Files\Octopus Deploy\Tentacle\Tentacle.exe register-with --instance Tentacle --server http://localhost:81 --name VAGRANT-2012-R2 --apiKey ********** --force --console --comms-style TentaclePassive --publicHostName localhost --environment "Env1" --environment "Env Number 2" --role "Role1" --role "Role 2"
C:\Windows\system32\sc.exe config "OctopusDeploy Tentacle" start= auto
C:\Program Files\Octopus Deploy\Tentacle\Tentacle.exe service --stop --instance Tentacle --console
C:\Program Files\Octopus Deploy\Tentacle\Tentacle.exe service --start --instance Tentacle --console

# change to polling
C:\Program Files\Octopus Deploy\Tentacle\Tentacle.exe deregister-from --server http://localhost:81 --apikey ********** --console --instance Tentacle
C:\Program Files\Octopus Deploy\Tentacle\Tentacle.exe configure --instance Tentacle --port 10943 --noListen True --console
C:\Program Files\Octopus Deploy\Tentacle\Tentacle.exe register-with --instance Tentacle --server http://localhost:81 --name VAGRANT-2012-R2 --apiKey ********** --force --console --comms-style TentacleActive --server-comms-port 10943 --environment "Env Number 2" --role "Role 2"
C:\Windows\system32\sc.exe config "OctopusDeploy Tentacle" start= auto
C:\Program Files\Octopus Deploy\Tentacle\Tentacle.exe service --stop --instance Tentacle --console
C:\Program Files\Octopus Deploy\Tentacle\Tentacle.exe service --start --instance Tentacle --console

# change to listening
C:\Program Files\Octopus Deploy\Tentacle\Tentacle.exe configure --instance Tentacle --port 10999 --console
C:\Program Files\Octopus Deploy\Tentacle\Tentacle.exe register-with --instance Tentacle --server http://localhost:81 --name VAGRANT-2012-R2 --apiKey ********** --force --console --comms-style TentaclePassive --publicHostName flooble.example.com --environment "Env Number 2" --role "Role 2"
C:\Windows\system32\sc.exe config "OctopusDeploy Tentacle" start= auto
C:\Program Files\Octopus Deploy\Tentacle\Tentacle.exe service --stop --instance Tentacle --console
C:\Program Files\Octopus Deploy\Tentacle\Tentacle.exe service --start --instance Tentacle --console
```

The Server thinks its a listening Tentacle.
The Tentacle thinks its both a polling and a listening Tentacle:
![image](https://cloud.githubusercontent.com/assets/373389/23523177/01a2045e-ffda-11e6-8854-40a30bcdbbce.png)

The Tentacle is not listening on the port. The log files say:
```
2017-03-03 05:25:10.4562      1  INFO  Waiting for service to start. Current status: Running
2017-03-03 05:25:10.6604      7  INFO  Agent will trust Octopus servers with the thumbprint: 596629F807F82E94B0A270EFA1873CE88D54E74E
2017-03-03 05:25:10.6604      7  INFO  Agent will trust Octopus servers with the thumbprint: 596629F807F82E94B0A270EFA1873CE88D54E74E
2017-03-03 05:25:10.6604      7  INFO  Agent will poll Octopus server at https://localhost:10943/ for subscription poll://ylhkccljmjpceqfcw5zk/ expecting thumbprint 596629F807F82E94B0A270EFA1873CE88D54E74E
2017-03-03 05:25:10.6604      7  INFO  Agent configured to use the system proxy, but no system proxy is configured for https://localhost:10943/
2017-03-03 05:25:10.6713      7  INFO  Agent will not listen on any TCP ports
2017-03-03 05:25:10.6713      7  INFO  The Windows Service has started
```

The config file has:
```
  <set key="Tentacle.Communication.TrustedOctopusServers">[{"Thumbprint":"596629F807F82E94B0A270EFA1873CE88D54E74E","CommunicationStyle":2,"Address":"https://localhost:10943","Squid":null,"SubscriptionId":"poll://ylhkccljmjpceqfcw5zk/"},{"Thumbprint":"596629F807F82E94B0A270EFA1873CE88D54E74E","CommunicationStyle":1,"Address":null,"Squid":null,"SubscriptionId":null}]</set>
  <set key="Tentacle.Deployment.ApplicationDirectory">C:\Applications</set>
  <set key="Tentacle.Services.NoListen">true</set>
  <set key="Tentacle.Services.PortNumber">10999</set>
``` 

So:
* config file is not being updated when `deregister-from` is called.
* I should be calling `deregister-from` when changing from polling to listening (currently not happening due to config file not being updated)
* I'm setting `noListen` to `true` on the polling Tentacle, and not re-setting it to `true` on the listening Tentacle. Should this default to `true` if you configure in `TentaclePassive` mode? If not, we should probably log some more detail saying that we've configured as listening but told it not to listen.

**Comment by matt-richardson** [comment]
Related to #3163

**Comment by NickJosevski** [comment]
Will get resolved by #3163

**Comment by NickJosevski** [comment]
Fixed by #3163
